Applications/MaterialAssist/Client/MaterialAssistClientBoards.cs

```diff
@@ -0,0 +1,196 @@
+    /// </summary>
+    public class MaterialAssistClientBoards : ServiceBase, IMaterialAssistClientBoards
+    {
+        /// <inheritdoc />
+        public MaterialAssistClientBoards(HttpClient client) : base(client) { }
+
+        #region Delete
+
+        /// <inheritdoc />
+        public async Task DeleteBoardEntity(string id)
+        {
+            var url = $"{_BaseRoute}/?{_Id}={Uri.EscapeDataString(id)}";
+
+            await DeleteObject(new Uri(url, UriKind.Relative));
+        }
+
+        /// <inheritdoc />
+        public async Task DeleteBoardEntities(IEnumerable<string> ids)
+        {
+            var url = $"{_BaseRoute}";
+
+            var boardCodes = new StringBuilder("?");
+            boardCodes.Append(string.Join("&", ids.Select(id => $"{_Id}={Uri.EscapeDataString(id)}")));
+
+            url += boardCodes;
+
+            await DeleteObject(new Uri(url, UriKind.Relative));
+        }
+
+        #endregion Delete
+
+        #region Constants
+
+        private const string _BaseRoute = "api/materialAssist/boardEntities";
+        private const string _BaseRouteMaterialManager = "api/materialManager/boards";
+        private const string _Id = "Id";
+        private const string _BoardCode = "boardCode";
+        private const string _BoardTypeCode = "boardTypeCode";
+        private const string _StorageLocation = "storageLocation";
+        private const string _DeleteFromInventory = "deleteFromInventory";
+        private const string _Length = "length";
+        private const string _Width = "width";
+        private const string _Comments = "comments";
+        private const string _RemovalType = "removalType";
+        private const string _Quantity = "quantity";
+
+        #endregion Constants
+
+        #region Read
+
+        /// <inheritdoc />
+        public async Task<IEnumerable<BoardEntity>> GetBoardEntities(int take, int skip = 0)
+        {
+            var url = $"{_BaseRoute}?take={take}&skip={skip}";
+
+            return await RequestEnumerable<BoardEntity>(new Uri(url, UriKind.Relative));
+        }
+
+        /// <inheritdoc />
+        public async Task<BoardEntity> GetBoardEntityById(string id)
+        {
+            var url = $"{_BaseRoute}?{_Id}={Uri.EscapeDataString(id)}";
+
+            return await RequestObject<BoardEntity>(new Uri(url, UriKind.Relative));
+        }
+
+        /// <inheritdoc />
+        public async Task<IEnumerable<BoardEntity>> GetBoardEntitiesByIds(IEnumerable<string> ids)
+        {
+            var url = $"{_BaseRoute}";
+
+            var boardCodes = new StringBuilder("?");
+            boardCodes.Append(string.Join("&", ids.Select(id => $"{_Id}={Uri.EscapeDataString(id)}")));
+
+            url += boardCodes;
+
+            return await RequestEnumerable<BoardEntity>(new Uri(url, UriKind.Relative));
+        }
+
+        /// <inheritdoc />
+        public async Task<IEnumerable<BoardEntity>> GetBoardEntitiesByBoardCode(string boardCode)
```

Comment:
GetBoardEntitiesByMaterialCode would also be interesting.

Reply:
We can add that of course